pkg/distributor/ha_tracker.go

```diff
@@ -211,9 +217,15 @@ func (c *haTracker) checkKVStore(ctx context.Context, key, replica string, now t
 	return c.client.CAS(ctx, key, func(in interface{}) (out interface{}, retry bool, err error) {
 		if desc, ok := in.(*ReplicaDesc); ok {
 
+			// Adjust update timeout slightly to spread heartbeats out over time with +/- UpdateTimeoutJitter
+			var jitter time.Duration
+			if c.cfg.UpdateTimeoutJitter != 0 {
+				jitter = time.Duration(rand.Int63n(int64(2*c.cfg.UpdateTimeoutJitter))) - c.cfg.UpdateTimeoutJitter
```

Comment:
I feel like we might be able to set the jitter for the distributor instead of per request. At least, I'd like to see what the write load reduction is doing it that way. @gouthamve thoughts?